diff --git a/usr/local/www/diag_packet_capture.php b/usr/local/www/diag_packet_capture.php
index 69fd677..91d61cd 100644
--- a/usr/local/www/diag_packet_capture.php
+++ b/usr/local/www/diag_packet_capture.php
@@ -265,6 +265,9 @@ include("head.inc"); ?>
 						<textarea style="width:98%" name="code" rows="15" cols="66" wrap="off" readonly="readonly">
 						<?php
 						system ("/usr/sbin/tcpdump $disabledns $detail -r $fp$fn");?>
+
+						conf_mount_ro();
+
 						</textarea><?php
 					}
 				}?>
@@ -277,8 +280,6 @@ include("head.inc"); ?>
 </td></tr></table>
 <?php 
 
-conf_mount_ro();
-
 include("fend.inc"); 
 
 ?>
