--- diag_packet_capture.php.orig	2010-08-05 18:33:38.000000000 +0000
+++ diag_packet_capture.php	2010-08-05 18:40:24.000000000 +0000
@@ -266,6 +266,7 @@
 						<?php
 						system ("/usr/sbin/tcpdump $disabledns $detail -r $fp$fn");?>
 						</textarea><?php
+						conf_mount_ro();
 					}
 				}?>
 				</td>
@@ -277,8 +278,6 @@
 </td></tr></table>
 <?php 
 
-conf_mount_ro();
-
 include("fend.inc"); 
 
 ?>
