Index: globals.inc
===================================================================
RCS file: /cvsroot/pfSense/etc/inc/globals.inc,v
retrieving revision 1.40.2.16
diff -u -r1.40.2.16 globals.inc
--- globals.inc	27 Feb 2007 20:45:31 -0000	1.40.2.16
+++ globals.inc	9 Sep 2007 20:54:52 -0000
@@ -47,6 +47,8 @@
     "cf_conf_path" => "/cf/conf",
     "www_path" => "/usr/local/www",
     "xml_rootobj" => "pfsense",
+	"admin_group" => "admins",
+	"product_name" => "pfSense",
     "pppoe_interface" => "ng0",
     "n_pptp_units" => 16,  /* this value can be overriden in pptp->n_pptp_units */
     "pptp_subnet" => 28,   /* this value can be overriden in pptp->pptp_subnet  */
