diff --git a/etc/inc/services.inc b/etc/inc/services.inc
index d014b13..09b1ae8 100644
--- a/etc/inc/services.inc
+++ b/etc/inc/services.inc
@@ -636,7 +636,7 @@ function services_dnsmasq_configure() {
 		//}
 
 		/* run dnsmasq */
-		mwexec("/usr/local/sbin/dnsmasq --all-servers {$args}");
+		mwexec("/usr/local/sbin/dnsmasq --all-servers --edns-packet-max=4096 {$args}");
 
 		if ($g['booting'])
 			echo "done.\n";
