Index: usr/local/www/fbegin.inc
===================================================================
RCS file: /cvsroot/pfSense/usr/local/www/fbegin.inc,v
retrieving revision 1.104.2.37.2.7
diff -u -r1.104.2.37.2.7 fbegin.inc
--- usr/local/www/fbegin.inc	31 May 2007 03:21:59 -0000	1.104.2.37.2.7
+++ usr/local/www/fbegin.inc	8 Sep 2007 21:16:29 -0000
@@ -127,6 +127,7 @@
 						<?php endif; ?>
 							<li><a href="/wizard.php?xml=setup_wizard.xml" class="navlnk">Setup wizard</a></li>
 							<li><a href="/system_routes.php" class="navlnk">Static routes</a></li>
+                            <?php echo return_ext_menu("System"); ?>
 						</ul>
 					</li>
 					<li class="drop">
